web/src/user_profile.js

```diff
@@ -194,6 +196,7 @@ function render_manage_profile_content(user) {
     const $container = $("#manage-profile-tab");
     $container.empty();
     settings_users.show_edit_user_info_modal(user.user_id, $container);
+    $("#user-profile-modal .modal__footer").show();
```

Comment:
Is there a reason you didn't do the hide/show operations here consistently in the `switch` statement below? That would be way easier to verify.

Reply:
This seems to work, added it:
``` diff
commit ef0f735472193fffdad9454baff83e3cb98d80c9 (HEAD -> main)
Author: Tim Abbott <tabbott@zulip.com>
Date:   Sun Sep 10 16:53:13 2023 -0700

    user_profile: Centralize footer logic.

diff --git a/web/src/user_profile.js b/web/src/user_profile.js
index ef8441af2b..db4fcd7776 100644
--- a/web/src/user_profile.js
+++ b/web/src/user_profile.js
@@ -158,7 +158,6 @@ function format_user_group_list_item(group) {
 }
 
 function render_user_stream_list(streams, user) {
-    $("#user-profile-modal .modal__footer").hide();
     streams.sort(compare_by_name);
     const $container = $("#user-profile-modal .user-stream-list");
     $container.empty();
@@ -179,7 +178,6 @@ function render_user_stream_list(streams, user) {
 }
 
 function render_user_group_list(groups, user) {
-    $("#user-profile-modal .modal__footer").hide();
     groups.sort(compare_by_name);
     const $container = $("#user-profile-modal .user-group-list");
     $container.empty();
@@ -206,7 +204,6 @@ function render_manage_profile_content(user) {
     } else {
         settings_users.show_edit_user_info_modal(user.user_id, $container);
     }
-    $("#user-profile-modal .modal__footer").show();
 }
 
 export function get_custom_profile_field_data(user, field, field_types) {
@@ -363,13 +360,13 @@ export function show_user_profile(user, default_tab_key = "profile-tab") {
         callback(_name, key) {
             $(".tabcontent").hide();
             $(`#${CSS.escape(key)}`).show();
+            $("#user-profile-modal .modal__footer").hide();
             $("#user-profile-modal .modal__body").removeClass("modal__body__manage_profile_height");
             $("#user-profile-modal .manage-profile-tab-footer").removeClass(
                 "modal__footer_wrapper",
             );
             switch (key) {
                 case "profile-tab":
-                    $("#user-profile-modal .modal__footer").hide();
                     initialize_user_type_fields(user);
                     break;
                 case "user-profile-groups-tab":
@@ -382,6 +379,7 @@ export function show_user_profile(user, default_tab_key = "profile-tab") {
                     render_user_stream_list(user_streams, user);
                     break;
                 case "manage-profile-tab":
+                    $("#user-profile-modal .modal__footer").show();
                     render_manage_profile_content(user);
                     break;
             }
```